cli/cmd/uninstall/uninstall.go

```diff
@@ -0,0 +1,439 @@
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	FlagAutoApprove    = "auto-approve"
+	DefaultAutoApprove = false
+
+	FlagNamespace        = "namespace"
+	DefaultAllNamespaces = ""
+
+	FlagReleaseName       = "name"
+	DefaultAnyReleaseName = ""
+
+	FlagWipeData    = "wipe-data"
+	DefaultWipeData = false
+
+	FlagSkipWipeData    = "skip-wipe-data"
+	DefaultSkipWipeData = false
+)
+
+type Command struct {
+	*common.BaseCommand
+
+	kubernetes kubernetes.Interface
+
+	set *flag.Sets
+
+	flagNamespace    string
+	flagReleaseName  string
+	flagAutoApprove  bool
+	flagWipeData     bool
+	flagSkipWipeData bool
+
+	flagKubeConfig  string
+	flagKubeContext string
+
+	once sync.Once
+	help string
+}
+
+func (c *Command) init() {
+	c.set = flag.NewSets()
+	{
+		f := c.set.NewSet("Command Options")
+		f.BoolVar(&flag.BoolVar{
+			Name:    FlagAutoApprove,
+			Target:  &c.flagAutoApprove,
+			Default: DefaultAutoApprove,
+			Usage:   "Skip approval prompt for uninstalling Consul.",
+		})
+		// This is like the auto-approve to wipe all data without prompting for non-interactive environments that want
+		// to remove everything.
+		f.BoolVar(&flag.BoolVar{
+			Name:    FlagWipeData,
+			Target:  &c.flagWipeData,
+			Default: DefaultWipeData,
+			Usage:   "Delete all PVCs, Secrets, and Service Accounts associated with Consul Helm installation without prompting for approval to delete. Only use this when persisted data from previous installations is no longer necessary.",
+		})
+		// This is like the auto-approve to NOT wipe all data without prompting for non-interactive environments that
+		// only want to remove the Consul Helm installation but keep the data.
+		f.BoolVar(&flag.BoolVar{
+			Name:    FlagSkipWipeData,
+			Target:  &c.flagSkipWipeData,
+			Default: DefaultSkipWipeData,
+			Usage:   "Skip deleting all PVCs, Secrets, and Service Accounts associated with Consul Helm installation without prompting for approval to delete.",
+		})
+		f.StringVar(&flag.StringVar{
+			Name:    FlagNamespace,
+			Target:  &c.flagNamespace,
+			Default: DefaultAllNamespaces,
+			Usage:   "Namespace for the Consul installation.",
+		})
+		f.StringVar(&flag.StringVar{
+			Name:    FlagReleaseName,
+			Target:  &c.flagReleaseName,
+			Default: DefaultAnyReleaseName,
+			Usage:   "Name of the installation. This will be prefixed to resources installed on the cluster.",
+		})
+
+		f = c.set.NewSet("Global Options")
+		f.StringVar(&flag.StringVar{
+			Name:    "kubeconfig",
+			Aliases: []string{"c"},
+			Target:  &c.flagKubeConfig,
+			Default: "",
+			Usage:   "Path to kubeconfig file.",
+		})
+		f.StringVar(&flag.StringVar{
+			Name:    "context",
+			Target:  &c.flagKubeContext,
+			Default: "",
+			Usage:   "Kubernetes context to use.",
+		})
+	}
+
+	c.help = c.set.Help()
+
+	// c.Init() calls the embedded BaseCommand's initialization function.
+	c.Init()
+}
+
+func (c *Command) Run(args []string) int {
+	c.once.Do(c.init)
+
+	defer func() {
+		if err := c.Close(); err != nil {
```

Comment:
Just out of curiousity what does c.Close() here do? would c.UI.Output() still be usable if c.Close() failed?

Reply:
I switched this to using the logger instead. In our case, the UI doesn't implement the closer interface, so Close() would be a no-op. But we could potentially choose to implement it in the future if we say "upgrade" our UI.